### PR TITLE
Handle static link to QCA ossl plugin on static builds

### PR DESCRIPTION
**Pull request by nirvn**
This PR fixes the authentication manager on Android, which had regressed since our move to a vcpkg static build setup.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 7cc246d879175ed006dae1f19f6ecdf9a1150641: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3313-8d670b97b737aa80be300bc46fb67b1a23467343-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3313-8d670b97b737aa80be300bc46fb67b1a23467343-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3313-8d670b97b737aa80be300bc46fb67b1a23467343-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3313-8d670b97b737aa80be300bc46fb67b1a23467343-x86-android.apk)
</details>

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +92,10 @@ function(create_executable)
     qt5_import_plugins(${exe_TARGET} INCLUDE ${QT_PKG}::QSvgPlugin)
   endif()
 
+  if(WITH_STATIC_QCA_PLUGINS)
```
[the remark this answers is not in the document]
Comment (nirvn) [reply src/app/CMakeLists.txt:95]: @m-kuhn , great idea

**Comment by nirvn** [comment]
@m-kuhn , it doesn't work unfortunately

**Comment by nirvn** [comment]
@m-kuhn , squashed and reverted to working solution.

File: src/app/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-kuhn) [reply src/app/CMakeLists.txt:95]: Let me check what we can do here. This should not appear in our public cmake interface

**Comment by nirvn** [comment]
@m-kuhn , I've cleaned up the commit, renamed the variable to HAVE_STATIC_QCA_PLUGINS (to match naming pattern in QGIS).

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**